### PR TITLE
Allow multiple class base types; intersect them, as with interfaces

### PR DESCRIPTION
**Pull request by weswigham**
Port #23123 to `release-2.8`

### CODE REVIEW

File: tests/cases/compiler/classMergedWithInterfaceMultipleBasesNoError.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+interface Bar { }
+interface Baz { }
+interface Q { }
+interface Foo extends Bar, Baz { }
+class Foo { }
+
+export default class extends Foo {
+    readonly observer = this.handleIntersection;
```
Comment (Kovensky) [inline tests/cases/compiler/classMergedWithInterfaceMultipleBasesNoError.ts]: This should still raise a use before define error (maybe it's a `--strict` thing?), but I guess it should still compile at any rate.

Comment (weswigham) [reply tests/cases/compiler/classMergedWithInterfaceMultipleBasesNoError.ts]: There's an [error baseline](https://github.com/weswigham/TypeScript/blob/11373d4e8e9b9fce6f8800d1b0788748d8d909f2/tests/baselines/reference/classMergedWithInterfaceMultipleBasesNoError.errors.txt)

Comment (Kovensky) [reply tests/cases/compiler/classMergedWithInterfaceMultipleBasesNoError.ts]: Oh, I missed it (the `NoError` threw me off) 🙇‍♀️

Comment (weswigham) [reply tests/cases/compiler/classMergedWithInterfaceMultipleBasesNoError.ts]: Kinda meant no error as in doesn't crash. Maybe not the clearest term.

**Comment by mihailik** [comment]
Does it mean what it appears to be saying?

Multiple inheritance?? As in:

```ts
class Base1 { color: string }
class Base2 { size: number }
class Derived extends Base1, Base2 { }
```

**Comment by weswigham** [comment]
Not multiple runtime inheritance, no. Just fixing multiple interface inheritance when said interface is merged with a class.

**Pull request merged**